lib/network/kmip/ClusterClient.ts

```diff
@@ -56,25 +90,168 @@ export default class ClusterClient implements KMSInterface {
         return this.clients[this.roundRobinIndex++];
     }
 
+    checkUnhealthyClient(unhealthyClient: UnhealthyClient) {
+        const client = unhealthyClient.client;
+        const healthyIndex = this.clients.indexOf(client);
+        if (healthyIndex === -1) {
+            this.clients.push(client);
+        } else {
+            // should not happen
+            this.logger.warn('checkUnhealthyClient: already moved in healthy', { unhealthyHost: client.host });
```

Comment:
What does mean, "already moved in healthy"

Reply:
the unhealthy client has already been pushed into the list of healthy clients that can be used in the round robin to send KMIP requests

Reply:
Okay got my answer in the next comment,.